### PR TITLE
feat: configure shoot worker node calico port mtu

### PR DESCRIPTION
**Pull request by jiangytcn**
**What this PR does / why we need it**:
let user configure the mtu for shoot worker node
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
NONE

```noteworthy user
If you wish to configure the MTU value of calico then you can set the `.spec.vethMTU` field in the `calico.networking.extensions.gardener.cloud/v1alpha1.NetworkConfig`. Please make sure that all the MTUs in your environment, failure to do so will result in performance penalties and packet loss.
```

### CODE REVIEW

File: controllers/networking-calico/pkg/apis/calico/v1alpha1/types_network.go [hunk 1]
```diff
@@ -79,6 +79,9 @@ type NetworkConfig struct {
 	// Typha settings to use for calico-typha component
 	// +optional
 	Typha *Typha `json:"typha,omitempty"`
+	// VethMTU settings used to configure calico port mtu
+	// +optional
+	VethMTU *string `json:"veth_mtu,omitempty"`
```
Comment (rfranzke) [inline controllers/networking-calico/pkg/apis/calico/v1alpha1/types_network.go]:
```suggestion
	VethMTU *string `json:"vethMTU,omitempty"`
```

File: controllers/networking-calico/docs/usage-as-end-user.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ kind: NetworkConfig
 ipam:
   type: host-local
   cidr: usePodCIDR
+veth_mtu: 1440
```
Comment (rfranzke) [inline controllers/networking-calico/docs/usage-as-end-user.md]:
```suggestion
vethMTU: 1440
```

File: controllers/networking-calico/docs/usage-as-end-user.md [hunk 3]
```diff
@@ -72,6 +73,7 @@ spec:
       kind: NetworkConfig
       ipam:
         type: host-local
+      veth_mtu: 1440
```
Comment (rfranzke) [inline controllers/networking-calico/docs/usage-as-end-user.md]:
```suggestion
      vethMTU: 1440
```

**Comment by jiangytcn** [comment]
thanks @rfranzke squash and rebased the branch, also fixed other field name to `vethMTU`

File: controllers/networking-calico/charts/internal/calico/templates/node/daemonset-calico-node.yaml [hunk 4]
```diff
@@ -76,7 +76,7 @@ spec:
             valueFrom:
               configMapKeyRef:
                 name: calico-config
-                key: veth_mtu
+                key: vethMTU
```
Comment (rfranzke) [inline controllers/networking-calico/charts/internal/calico/templates/node/daemonset-calico-node.yaml]:
```suggestion
                key: veth_mtu
```

File: controllers/networking-calico/charts/internal/calico/templates/node/daemonset-calico-node.yaml [hunk 5]
```diff
@@ -140,7 +140,7 @@ spec:
               valueFrom:
                 configMapKeyRef:
                   name: calico-config
-                  key: veth_mtu
+                  key: vethMTU
```
Comment (rfranzke) [inline controllers/networking-calico/charts/internal/calico/templates/node/daemonset-calico-node.yaml]:
```suggestion
                  key: veth_mtu
```

File: controllers/networking-calico/charts/internal/calico/templates/node/configmap-calico-config.yaml [hunk 6]
```diff
@@ -16,7 +16,7 @@ data:
   # Configure the Calico backend to use.
   calico_backend: "{{ .Values.config.backend }}"
   # Configure the MTU to use
-  veth_mtu: "1440"
+  vethMTU: "{{ .Values.config.vethMTU }}"
```
Comment (rfranzke) [inline controllers/networking-calico/charts/internal/calico/templates/node/configmap-calico-config.yaml]:
```suggestion
  veth_mtu: "{{ .Values.config.vethMTU }}"
```

This is the calico configmap, so I think at least the key here must be `veth_mtu`, right?

File: controllers/networking-calico/charts/internal/calico/values.yaml [hunk 7]
```diff
@@ -1,6 +1,7 @@
 global:
   podCIDR: ""
 config:
+  vethMTU: 1440
```
Comment (rfranzke) [inline controllers/networking-calico/charts/internal/calico/values.yaml]:
```suggestion
  veth_mtu: 1440
```

File: controllers/networking-calico/pkg/charts/utils.go [hunk 8]
```diff
@@ -36,6 +37,7 @@ type calicoConfig struct {
 	IPv4    ipv4                   `json:"ipv4"`
 	IPAM    ipam                   `json:"ipam"`
 	Typha   typha                  `json:"typha"`
+	VethMTU string                 `json:"vethMTU"`
```
Comment (rfranzke) [inline controllers/networking-calico/pkg/charts/utils.go]:
```suggestion
	VethMTU string                 `json:"veth_mtu"`
```

File: controllers/networking-calico/charts/internal/calico/templates/node/configmap-calico-config.yaml [hunk 9]
```diff
@@ -16,7 +16,7 @@ data:
   # Configure the Calico backend to use.
   calico_backend: "{{ .Values.config.backend }}"
   # Configure the MTU to use
-  veth_mtu: "1440"
+  veth_mtu: "{{ .Values.config.vethMTU }}"
```
Comment (rfranzke) [inline controllers/networking-calico/charts/internal/calico/templates/node/configmap-calico-config.yaml]:
```suggestion
  veth_mtu: "{{ .Values.config.veth_mtu }}"
```

**Comment by rfranzke** [comment]
@jiangytcn the tests are failing now

**Comment by jiangytcn** [comment]
@rfranzke sorry about that, fixed the test suits wording issue.

**Pull request merged**